app/views/admin/sponsors/edit.html.haml

```diff
@@ -0,0 +1,38 @@
+.row
+  .col-md-12
+    .page-header
+      %h1
+        Edit
+        = @sponsor.name
+.row
+  .col-md-8
+    = semantic_form_for(@sponsor, url: admin_conference_sponsor_path(@conference.short_title, @sponsor)) do |f|
+      = f.input :name
+      = f.input :amount, as: :number, label: 'Donation amount'
+      = f.input :description, input_html: { rows: 5, data: { provide: 'markdown-editable' } }
+      = f.input :picture
+      = f.input :website_url
+      = f.input :sponsorship_level, collection: @conference.sponsorship_levels
+      = f.input :address, label: 'Sponsor\'s address', hint: 'Please include the legal company name, along with the address, as you want it to be shown on the invoice'
+      = f.input :vat, label: 'VAT Registration Number'
+      %b Has Swags?
+      = check_box_tag @conference.short_title, @sponsor.id, @sponsor.has_swag,
+      method: :patch, url: "/admin/conferences/#{@conference.short_title}/sponsors/#{@sponsor.id}?sponsor[has_swag]=",
+      class: 'switch-checkbox', data: { size: 'small',
+                                        off_color: 'warning',
+                                        on_text: 'Yes',
+                                        off_text: 'No' }
+      - @sponsor.swag.each_with_index do |(key, value), index|
+        = render partial: 'swag_fields', locals: {f: f, index: index, type_value: value[:type].to_s, quantity_value: value[:quantity].to_i}
```

Comment:
value here is the hash `{type: 'xyz', quantity: 10}` right?
If so, then why don't we simply pass this hash as parameter to the partial file, and get the type and quantity inside of the partial?
It will simplify our view, and it will make our code easier to read and understand.

Reply:
Sure, but this isn't working when the form is rendered for new, empty fields. The partial doesn't recognize value as a hash, thus an error occurs. This is why I preferred to use `type_value` and `quantity_value`.

Reply:
I have already tried to work on it but it turned out to be more challenging than I expected